sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -141,14 +141,16 @@ object ExtractEquiJoinKeys extends Logging with PredicateHelper {
 }
 
 /**
- * A pattern that collects the filter and inner joins.
+ * A pattern that collects the filter and inner joins and skip projections with attributes only.
  *
  *          Filter
  *            |
  *        inner Join
  *          /    \            ---->      (Seq(plan0, plan1, plan2), conditions)
  *      Filter   plan2
  *        |
+ *      Project
```

Comment:
I don't think this case would be covered, i.e., Project-over-Filter or Filter-over-Project.

Reply:
yea, good catch. Thanks! I'll fix soon.

Reply:
I fixed the bug you pointed out and added the two tests for that. Could you check again? Thanks!
https://github.com/apache/spark/pull/20345/files#diff-6c81baa433aea4741c60c00734f31406R191